### ISSUE TITLE
google_checks.xml: false negative 3.2 Package declaration

### ISSUE THREAD

**Issue by Atharv3221** [checkstyle/checkstyle#21444]
Child Issue of #21428

I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

[3.2 Package declaration](https://google.github.io/styleguide/javaguide.html#s3.2-package-declaration)

<img width="938" height="163" alt="Image" src="https://github.com/user-attachments/assets/36f2d5ff-c9e5-40d6-8a1d-b28c41a64706" />

## missing package declaration
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
/** Some javadoc. */
public class Test {
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c /google_checks.xml Test.java
Starting audit...
Audit done.
```
Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:2:1: Missing package declaration. [PackageDeclaration]
Audit done.
Checkstyle ends with 1 errors.
```

## Use of compact source file
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat Test.java
void main() {
  int a = 2;
}
```
```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.1.0-all.jar -c /google_checks.xml Test.java
Starting audit...
Audit done.
```
Expected:
```
Starting audit...
[ERROR] /home/atharv/test/Test.java:1:1: Compact source files are not allowed. [MatchXpath]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
>missing package declaration

Lets try to do it by some regexp or xpath.
In real life it is not possible to happen, as sources will not compile if package is missing. So it is realistic to be missing only in compact sources.

**Comment by Atharv3221** [comment]
I am working on it

**Comment by romani** [comment]
Google config is updated
